packages/wdio-cucumber-framework/src/cucumberFormatter.ts

```diff
@@ -68,11 +69,11 @@ export default class CucumberFormatter extends Formatter {
         let results: TestStepResult[] = []
         options.eventBroadcaster.on('envelope', (envelope: Envelope) => {
             if (envelope.gherkinDocument) {
-                this.onGherkinDocument(envelope.gherkinDocument)
+                this.onGherkinDocument({ ...envelope.gherkinDocument, uri: this.normalizeURI(envelope.gherkinDocument.uri) })
             } else if (envelope.testRunStarted) {
                 this.onTestRunStarted()
             } else if (envelope.pickle) {
-                this.onPickleAccepted(envelope.pickle)
+                this.onPickleAccepted({ ...envelope.pickle, uri: this.normalizeURI(envelope.pickle.uri)!! })
```

Comment:
The reason I added `!!` is that `normalizeURI` should not be left as `undefined` for `envelope.pickle.uri`. Without this, it would cause issues with data types.

Reply:
This seems to me we just suppress the case where `normalizeURI` would return undefined. How about we change its interface to this:

```ts
    normalizeURI(uri: string) {
        return path.isAbsolute(uri) ? uri : path.resolve(uri)
    }
```

given that `envelope.pickle.uri` is always defined as `string`.

Reply:
I agree with your suggestion to modify the `normalizeURI` function as proposed above. I'll go ahead and implement those changes now.

Reply:
My apologies, GherkinDocument has a type `uri?: string`, and pickle has type as `uri: string`. This modification cause type conflicts with GherkinDocument.

Reply:
for the `onGherkinDocument` call we could do:
```ts
                this.onGherkinDocument({
                    ...envelope.gherkinDocument,
                    ...(envelope.gherkinDocument.uri
                        ? { uri: this.normalizeURI(envelope.gherkinDocument.uri) }
                        : {}
                    )
                })
```